packages/core/lib/generators/assets_generator.dart

```diff
@@ -49,6 +49,17 @@ class AssetsGenConfig {
       flutterGen.assets.outputs.packageParameterEnabled ? _packageName : '';
 }
 
+List<Object> _buildAssetsList(Config config) {
+  final List<Object> deferredAssets = [];
+  config.pubspec.flutter.deferredComponents?.forEach((deferredComponent) {
+    // Include all manipulated assets to the list of deferred assets.
+    deferredAssets.addAll(deferredComponent.assets ?? []);
+  });
+
+  // Merge the deferred assets with the main assets.
+  return [...config.pubspec.flutter.assets, ...deferredAssets];
```

Comment:
Wouldn't `[...?config.pubspec.flutter.deferredAssets?.assets]` do the same thing as above?

Reply:
That's true...! good catch!
BTW, do you mean? 

![image](https://github.com/user-attachments/assets/d52517fd-1300-4e14-b236-535e31d43491)

Reply:
Yeah exactly

Reply:
I'm sorry. For a moment I thought it was the same but it is not. I need to revert it.

The fact is we may have many Deferret Components, with many assets each (a list of lists).

So before spreading the list of deferredComponents, it is necessary to spread the assets from each deferredComponent

Reply:
So you may reduce components' assets into one?

Reply:
I didn't get that. 🤔
What do you mean by reduce components assets?

Reply:
Something like `reduce((list, assets) => list + assets)`

Reply:
I belive I'd need to use `fold` instead of reduce to have an initial empty list (I might have nothing at this point):
So I got something like this: 
![image](https://github.com/user-attachments/assets/38488c7c-905b-4f7f-afaf-d425450c56fd)

Reply:
any thoughts?

Reply:
@ianmaciel Looks good!

Reply:
here it is :)

5f4a617cc15a64d8483ee54befe0cdb57da6dfbe